transport/httptransport/http_transport.go

```diff
@@ -0,0 +1,235 @@
+package httptransport
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+
+	"github.com/google/uuid"
+
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/filecoin-project/boost/transport/types"
+
+	"github.com/filecoin-project/boost/transport"
+)
+
+var log = logging.Logger("http-transport")
+
+const (
+	// 1 Mb
+	readBufferSize = 1048576
+)
+
+var _ transport.Transport = (*httpTransport)(nil)
+
+func New() *httpTransport {
+	return &httpTransport{}
+}
+
+type httpTransport struct {
+	// TODO Construct and inject an http client here ?
+}
+
+func (h *httpTransport) Execute(ctx context.Context, transportInfo []byte, dealInfo *types.TransportDealInfo) (th transport.Handler, err error) {
+	// de-serialize transport opaque token
+	tInfo := &types.HttpRequest{}
+	if err := json.Unmarshal(transportInfo, tInfo); err != nil {
+		return nil, fmt.Errorf("failed to de-serialize transport info bytes: %w", err)
```

Comment:
nit: include the bytes as a string in the error message to make it easier to debug